op-node/rollup/async/asyncgossiper.go

```diff
@@ -0,0 +1,123 @@
+package async
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+)
+
+// AsyncGossiper is a component for gossiping a payload while running the derivation pipeline
+// the gossiping is done in a separate goroutine, so that the derivation pipeline can continue without waiting for the gossiping to finish
+// it also maintains the currently gossiping payload, so it can be checked against other found payloads
```

Comment:
I'm not sure what "checked against other found payloads" means here. The PR description made more sense to me - that we to try our best to avoid gossiping a valid block that we don't wind up importing for some reason.

Reply:
Yeah I lost the thread a bit there. Rewrote most of these comments through the whole file.